components/dashboard/RequestView.tsx

```diff
@@ -0,0 +1,304 @@
+import { useMemo, useState, ReactElement, useEffect } from "react";
+import { ButtonPrimary, ButtonSecondary } from "@/components/core/Button";
+import { DashboardCard } from "@/components/core/Layout";
+import { TabPrimary } from "@/components/core/Tabs";
+import NextLink from "next/link";
+import DashboardSearchInput from "@/components/core/DashboardSearchInput";
+import {
+  Link,
+  Tabs,
+  ButtonGroup,
+  TabList,
+  TabPanels,
+  TabPanel,
+  Table,
+  Thead,
+  Spacer,
+  Box,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  IconButton,
+  useDisclosure,
+  useToast,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  ModalFooter,
+  Modal,
+} from "@chakra-ui/react";
+import { PropertyImage } from "@/components/sections/Listings";
+import { TenantRequestEntry } from "@/src/api/types";
+import { auth } from "@/src/firebase";
+import { useAuthState } from "react-firebase-hooks/auth";
+import { tenantRequests } from "@/src/api/collections";
+import { useCollection } from "react-firebase-hooks/firestore";
+import { getUserPublicById } from "@/src/api/queries";
+import { DeleteIcon, CloseIcon, CheckIcon } from "@chakra-ui/icons";
+import {
+  listingRouteBuilder,
+  listingHrefBuilder,
+} from "@/src/utils/listingRoute";
+import { Heading5 } from "../core/Text";
+
+enum Request {
+  Sent = "tenantUid",
+  Received = "landlordUid",
+}
+
+interface DeleteProps {
+  isOpen: boolean;
+  onClose: () => void;
+  request: TenantRequestEntry | undefined;
+}
+
+function RequestView(): ReactElement {
+  // @ts-ignore
+  const [selectedRequest, setSelectedRequest] = useState<Request>(
+    Request.Received
+  );
+  const [tenantRequestList, setTenantRequestList] = useState<
+    TenantRequestEntry[]
+  >([]);
+
+  const [user] = useAuthState(auth);
+  const query = useMemo(
+    () =>
+      user ? tenantRequests.where(selectedRequest, "==", user.uid) : undefined,
+    [user, selectedRequest]
+  );
+
+  const [snapshot] = useCollection(query);
+
+  const tenantRequestPromise = snapshot?.docs.map(async (doc) => {
+    const requestData = doc.data();
+    const publicUserInfo = await getUserPublicById(
+      requestData[
+        selectedRequest === Request.Received ? "tenantUid" : "landlordUid"
+      ]
+    );
+    return {
+      ...doc.data(),
+      ...publicUserInfo,
+      id: doc.id,
+    } as TenantRequestEntry;
+  });
+
+  useEffect(() => {
+    const handleTenantRequestPromise = async () => {
+      if (tenantRequestPromise) {
+        setTenantRequestList(await Promise.all(tenantRequestPromise));
+      }
+    };
+
+    handleTenantRequestPromise();
+  }, [user, selectedRequest, snapshot, tenantRequestPromise]);
```

Comment:
An infinite re-render loop is triggered here, causing a ton of Firestore requests to be continuously sent (as seen in the network tab in dev tools) and lagging the page from all the updates. I think `tenantRequestPromise` being part of the dependency array and being remade each time outside the `useEffect` hook is the main issue.

Consider resolving by

1. Taking `tenantRequestPromise` out of the dependency array
2. Moving `tenantRequestPromise` in the `useEffect` hook
3. Optionally also remove `user` and `selectedRequest` out of the dependency array, leaving just `snapshot`, to keep it cleaner since those are not directly used in the hook

So it can look like the following:

```javascript
  // tenantRequestPromise is removed from above

  useEffect(() => {
    const handleTenantRequestPromise = async () => {
      const tenantRequestPromise = snapshot?.docs.map(async (doc) => {
        const requestData = doc.data();
        const publicUserInfo = await getUserPublicById(
          requestData[
            selectedRequest === Request.Received ? "tenantUid" : "landlordUid"
          ]
        );

        return {
          ...doc.data(),
          ...publicUserInfo,
          id: doc.id,
        } as TenantRequestEntry;
      });

      if (tenantRequestPromise) {
        setTenantRequestList(await Promise.all(tenantRequestPromise));
      }
    };

    handleTenantRequestPromise();
  }, [snapshot]);
```

Reply:
Them reads :eyes: 

![image](https://user-images.githubusercontent.com/20251243/142332365-4c5fd4d8-2e6a-4cba-b161-e4e9b1211232.png)

Reply:
oh shit LOOl

Reply:
fixed 7c9614d